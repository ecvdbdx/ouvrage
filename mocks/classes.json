[
  {
    "degree": "m1",
    "speciality": "dÃ©velopement",
    "avatar": "image.jpeg",
    "students": [1, 2, 3, 4, 5, 6, 7, 8],
    "id": "1",
    "year": "2017/2018"
  }
]
