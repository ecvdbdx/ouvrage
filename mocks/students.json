[
  {
    "username": "epatrois",
    "first_name": "Manu",
    "last_name": "Patrois",
    "id": "2"
  },
  {
    "id": "5",
    "username": "mdewaele",
    "first_name": "Maxence",
    "last_name": "De Waele",
    "email": "maxencedewaele@gmail.com",
    "birthdate": "838921463",
    "avatar": "https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png"

  },
  {
    "id": "8",
    "username": "rodrigodelanoche",
    "first_name": "Vincent",
    "last_name": "Rodriguez",
    "email": "vincentrodrigu@hotmail.fr",
    "birthdate": "14/05/1413",
    "avatar": "http://etc.usf.edu/clipart/5500/5596/columbus_14_md.gif"
  },
  {
    "id": "6",
    "username": "erozis",
    "first_name": "Emilie",
    "last_name": "Rozis",
    "email": "emilie.40@hotmail.com"
  },
  {
    "first_name": "Hubert",
    "username": "hmoncenis",
    "id": "1",
    "last_name": "Montcenis"
  },
  {
    "id": "7",
    "username": "vivianeglz",
    "first_name": "Viviane",
    "last_name": "Gonzalez",
    "email": "viviane.gonzalez@outlook.fr",
    "birthdate": "02/06/1995",
    "avatar": "kikou.png"
  },
  {
    "id": "3",
    "username": "antoinesinety",
    "first_name": "Antoine",
    "last_name": "de Sinety",
    "email": "fake@yopmail.fr",
    "birthdate": "08/06/1995",
    "avatar": "Image.png"
  },
  {
    "id": "4",
    "username": "tgauzin",
    "first_name": "Thomas",
    "last_name": "Gauzin",
    "email": "thomas.gauzin@gmail.com"
  }
]
