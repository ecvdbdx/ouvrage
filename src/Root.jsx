import React from 'react';

export default class Root extends React.Component {
  constructor () {
    super();
  }

  render () {
    return (
      <div>
        <h1>Ouvrage</h1>
      </div>
    );
  }
}
