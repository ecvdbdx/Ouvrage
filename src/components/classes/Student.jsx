import React, { Component } from 'react';

export default class Student extends Component {
  render () {
    return (
      <div className="content-student" onClick={() => this.props.goToProfile(this.props)}>
        <img className="avatar" src={this.props.avatar} />
        <span className="username">{ this.props.username }</span>
        <span className="lastname">{ this.props.lastName }</span>
        <span className="firstname">{ this.props.firstName }</span>
        <span className="email">{ this.props.email }</span>
      </div>
    );
  }
}

Student.defaultProps = {
  avatar: ['https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png']
};
