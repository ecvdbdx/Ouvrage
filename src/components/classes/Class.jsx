import React, { Component } from 'react';
import Student from './Student.jsx';


export default class Class extends Component {
  render () {
    var myStudents = this.props.students.map((item, index) => <Student key={index} goToProfile={this.props.studentClick}{...this.props.students[index]} />);
    return (
      <div>
        <h2 id="speciality">{ this.props.speciality }</h2>
        <div className="wrapper-students">{myStudents}</div>
      </div>
    );
  }
}
