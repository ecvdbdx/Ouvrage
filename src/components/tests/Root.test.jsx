import React from 'react';
import renderer from 'react-test-renderer';

import Root from '../../Root.jsx';
import classes from '../../../mocks/classes.json';
import students from '../../../mocks/students.json';

it('renders correctly', () => {
  const tree = renderer.create(<Root />).toJSON();
  expect(tree).toMatchSnapshot();
});
