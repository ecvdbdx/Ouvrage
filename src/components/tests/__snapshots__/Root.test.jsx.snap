// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header>
    <h1>
      Ouvrage - ECV Digital Bordeaux
    </h1>
    <p>
      Ouvrage
    </p>
  </header>
  <div>
    <button
      onClick={[Function]}
    >
      Accueil
    </button>
    <button
      onClick={[Function]}
    >
      Accéder au chat
    </button>
    <button
      onClick={[Function]}
    >
      Mon profil
    </button>
    <button
      onClick={[Function]}
    >
      Créer un profil
    </button>
  </div>
  <div>
    <section>
      <header>
        <h2>
          L'ECV DIGITAL
        </h2>
      </header>
      <p>
        L’ECV Digital a été créée afin de combler les besoins du marché de l’emploi qui manque cruellement de profils web pour accompagner la transformation digitale de toutes les entreprises et de l’économie en général. Située à Paris, Nantes, Bordeaux et Aix-en-Provence, l’ECV Digital, l’école du numérique et du Web, forme à tous les métiers du digital. Issue du groupe ECV (École de Communication Visuelle), l’école se positionne comme une école généraliste du secteur internet. Son objectif est de former des profils polyvalents intégrant les compétences techniques, créatives et marketing.
      </p>
    </section>
    <section>
      <header>
        <h2>
          Voici la liste des classes
        </h2>
      </header>
      <ol
        className="grid-3"
      />
    </section>
  </div>
</div>
`;
